lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -776,13 +785,29 @@ bool ProxyRenderDelegate::getInstancedSelectionPath(
     }
 
     const Ufe::PathSegment pathSegment(usdPath.GetText(), USD_UFE_RUNTIME_ID, USD_UFE_SEPARATOR);
-    const Ufe::SceneItem::Ptr& si
-        = handler->createItem(_proxyShapeData->ProxyShape()->ufePath() + pathSegment);
+    Ufe::Path sceneItemPath(_proxyShapeData->ProxyShape()->ufePath() + pathSegment);
+
+    if (instanceIndex != UsdImagingDelegate::ALL_INSTANCES) {
+        const std::string& primName = sceneItemPath.back().string();
+        const Ufe::PathComponent primInstanceComponent(
+            TfStringPrintf("%s.instance_%d", primName.c_str(), instanceIndex));
+        sceneItemPath = sceneItemPath.sibling(primInstanceComponent);
```

Comment:
Here's where things start getting a little sketchy. It's seeming like we need a `Ufe::Path` that is unique to each particular instance. If we don't do this, when we add the scene items to the UFE selection below, it appears to be collapsing multiple items into one when they all share the same path, even though we're setting a different `instanceIndex` on each one. In `PopulateSelection()`, we only get a single `SceneItem` per path out of the global selection.

I'm intentionally only inventing this `PathComponent` on the UFE side to avoid polluting any tables on the USD side by creating `TfTokens` or `SdfPaths` there that we'll never use again. I would imagine though that it's a similar issue on the UFE side, so I'm a little concerned about the performance implications of having a `Ufe::Path` per instance. `PointInstancer` prims can easily reach instance counts into the hundreds of thousands or more.

This disconnect though between the `Ufe::Path` not exactly matching the `SdfPath` of the `UsdPrim` it represents could be problematic. For example, the outliner currently seems to get confused when an instance is selected. It highlights all ancestors of the `PointInstancer` using the "secondary" highlighting, but doesn't highlight the `PointInstancer` itself.

Reply:
Yes, Ufe::Selection will ignore duplicate items, i.e. items that compare equal.  Currently, item comparison is hard-coded to compare only paths.  We can explore the implications of changing this.

Yes, UFE has a problem similar to USD, in that Ufe::PathComponent is like a TfToken, a fixed, unmutable string stored in a table.  I can make a few observations here:
- Each PathComponent is a small object, and the strings for the integer indices are small, and stored once.
- If we have Ufe::PathComponents for an instancer, those strings will get re-used, won't they?
- In the UFE implementation, the table that maps strings to path components is typedef std::unordered_map<std::string, std::weak_ptr<Ufe::PathComponent::Data> > StringTable.  We may be able to garbage collect the table if a stored weak_ptr is expired.
- The Ufe::PathComponent::Data is opaque from the point of view of the PathComponent interface.  We may be able to play games there and have a union that stores an integer index.  This is starting to sound like a bit of a stretch, admittedly.
- The performance impact of having a Ufe::Path per instance would have to be measured.  Since each path component is the size of a shared_ptr, and assuming we have say 10 path component in a Ufe::Path, then we're talking the size of 10 x shared_ptr x 100000 instances, which doesn't sound so bad.  What other implications are you concerned about?
- We would have to figure out what the trouble with the Outliner is.